client-spark/spark3/src/main/java/org/apache/spark/shuffle/RssShuffleManager.java

```diff
@@ -1201,61 +1209,92 @@ public synchronized boolean reassignAllShuffleServersForWholeStage(
 
   // this is only valid on driver side that exposed to being invoked by grpc server
   @Override
-  public ShuffleServerInfo reassignFaultyShuffleServerForTasks(
-      int shuffleId, Set<Integer> partitionIds, String faultyShuffleServerId) {
+  public ShuffleHandleInfo reassignFaultyShuffleServerForTasks(
+      int shuffleId,
+      Set<Integer> partitionIds,
+      String faultyShuffleServerId,
+      Set<Integer> needLoadBalancePartitionIds) {
     ShuffleHandleInfo handleInfo = shuffleIdToShuffleHandleInfo.get(shuffleId);
     synchronized (handleInfo) {
       // find out whether this server has been marked faulty in this shuffle
       // if it has been reassigned, directly return the replacement server.
       // otherwise, it should request new servers to reassign
-      Set<ShuffleServerInfo> replacements =
-          handleInfo.getExistingReplacements(faultyShuffleServerId);
+      Set<ShuffleServerInfo> replacements = handleInfo.getReplacements(faultyShuffleServerId);
       if (replacements == null) {
-        replacements = requestServersForTask(shuffleId, partitionIds, faultyShuffleServerId);
+        int requiredServerNum = 1;
+        if (CollectionUtils.isNotEmpty(needLoadBalancePartitionIds)) {
+          requiredServerNum = partitionReassignLoadBalanceServerNum;
```

Comment:
`partitionReassignLoadBalanceServerNum` may dynamic value in future by calculate by partition size(not support now) or other logic, Is it better to set the `partitionReassignLoadBalanceServerNum` value in the RPC request?